fjord/analytics/templates/analytics/analyzer/hb_data.html

```diff
@@ -22,6 +22,19 @@
     <a class="first-link" href="{{ request.get_full_path()|urlparams(sortby=sortby, page=1) }}">First</a>
     |
     <a class="last-link" href="{{ request.get_full_path()|urlparams(sortby=sortby, page=answers.paginator.num_pages) }}">Last</a>
+
+    &mdash;
+
+    <form method="GET" class="inline-form" action="{{ request.get_full_path() }}">
+      <select name="survey">
+        <option value=""{% if not survey %} selected{% endif %}>ALL</option>
+        {% for obj in surveys %}
+          <option value="{{ obj.id }}"{% if survey and survey.id == obj.id %} selected{% endif %} >{{ obj.name }}</option>
+        {% endfor %}
+      </select>
+      <input type="hidden" name="sortby" value="{{ sortby }}">
```

Comment:
The page parameter gets tossed. And yeah, that's what I would expect when changing a filter (going back to page 1). :+1: